### PR TITLE
ls-lint: add lint:ls

### PR DESCRIPTION
**Pull request by himdel**
add [ls-lint](https://ls-lint.org/), a filename linter
and a `lint:ls` task that runs it during `npm run lint`

and a `.ls-lint.yml` config, with a ruleset that contains some of our current exceptions to pass without renames
we may want to change that and remove a couple of the rules + rename & update imports to match

---

We *may* want to remove some of the `| another way` bits and rename the files instead.

### CODE REVIEW

File: .ls-lint.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+ls:
+  .travis:
+    .js: snake_case
+
+  config:
+    .js: point.case | kebab-case
```
Comment (himdel) [inline .ls-lint.yml:6]: without `| kebab-case`:

```
2021/08/30 14:53:26 ./config/webpack-ts-overrides.js failed for rules: pointcase
```

File: .ls-lint.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src/loaders/insights:
+    .js: kebab-case | PascalCase
```
Comment (himdel) [inline .ls-lint.yml:9]: without `| PascalCase`

```
2021/08/30 14:54:10 ./src/loaders/insights/Routes.js failed for rules: kebabcase
```

File: .ls-lint.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  src/utilities:
+    .js: camelCase | snake_case
```
Comment (himdel) [inline .ls-lint.yml:12]: without `camelCase |`:

```
2021/08/30 14:55:47 ./src/utilities/asyncComponent.js failed for rules: snakecase
2021/08/30 14:55:47 ./src/utilities/getBaseName.js failed for rules: snakecase
```

File: .ls-lint.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  test:
+    .js: kebab-case | snake_case
```
Comment (himdel) [inline .ls-lint.yml:15]: without `| snake_case`:

```
2021/08/30 14:56:42 ./test/cypress/integration/api_token.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/docs_menu.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/group_management.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/group_permissions.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/namespace_form.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/repo_management.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/token_management.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/user_dashboard.js failed for rules: kebabcase
2021/08/30 14:56:42 ./test/cypress/integration/user_filter.js failed for rules: kebabcase
```

**Review by ZitaNemeckova — APPROVED** [review]
LGTM :+1: I take that having consistency in file names will come later :)

**Pull request merged**